app/javascript/packages/document-capture/components/in-person-location-post-office-search-step.tsx

```diff
@@ -77,15 +86,82 @@ const requestUspsLocations = async (address: LocationQuery): Promise<FormattedLo
   return formatLocation(response);
 };
 
-function InPersonLocationPostOfficeSearchStep({ onChange, toPreviousStep, registerField }) {
+const ADDRESS_SEARCH_URL = '/api/addresses';
+
+function requestAddressCandidates(unvalidatedAddressInput: string): Promise<Location[]> {
+  return request<Location[]>(ADDRESS_SEARCH_URL, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    json: { address: unvalidatedAddressInput },
+  });
+}
+
+function useUspsLocations() {
+  // raw text input that is set when user clicks search
+  const [addressQuery, setAddressQuery] = useState('');
+  const [fieldValidation, setFieldValidation] = useState<RefObject<HTMLFormElement>>();
```

Comment:
This should accept an object with interface methods `setCustomValidity` and `reportValidity`, not a full `HTMLFormElement`. Such a reference should only be obtained via `useImperativeHandle`, which should return a proxy object created inside of `AddressSearch`.